db/migrate/20231117182638_fill_approve_unless_allowed_groups_based_on_deprecated_settings.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+class FillApproveUnlessAllowedGroupsBasedOnDeprecatedSettings < ActiveRecord::Migration[7.0]
+  def up
+    approve_unless_trust_level_raw =
+      DB.query_single(
+        "SELECT value FROM site_settings WHERE name = 'approve_unless_trust_level'",
+      ).first
+
+    # Default for old setting is TL0, we only need to do anything if it's been changed in the DB.
+    if approve_unless_trust_level_raw.present?
+      # Matches Group::AUTO_GROUPS to the trust levels.
+      approve_unless_allowed_groups = "1#{approve_unless_trust_level_raw}"
```

Comment:
Should this be

```suggestion
      approve_unless_allowed_groups = "1|#{approve_unless_trust_level_raw}"
```

?

Reply:
Nope, It's just doing this under the hood https://github.com/discourse/discourse/blob/4346512f88ebf3fa4aa0cd6a68d98781fe70423d/db/migrate/20231024034031_migrate_tl_to_group_settings_anonymous_posting_min_tl.rb#L14-L25

It's just pretending a 1, so that trust level 0 becomes group ID "10", trust level 1 becomes group ID "11", etc.

Reply:
Oh this is clever, I wish I had thought of this!